pkg/webui/console/containers/device-importer/index.js

```diff
@@ -212,38 +310,125 @@ export default class DeviceImporter extends Component {
               pulse={status === 'processing'}
-              status={statusMap[status] || 'unknown'}
+              status={aborted ? 'mediocre' : statusMap[status] || 'unknown'}
             />
             <ProgressBar
-              current={devicesComplete}
-              target={totalDevices}
+              current={devicesProcessed}
+              target={convertedDevices.length}
               showStatus
-              showEstimation={!hasErrored}
+              showEstimation={!hasErrored && !aborted}
               className={style.progressBar}
-            />
-          </React.Fragment>
+            >
+              <Message
+                content={m.progress}
+                values={{
+                  errorCount: devicesProcessed - deviceErrors.length,
+                  deviceCount: convertedDevices.length,
+                }}
+              />
+            </ProgressBar>
+            {status === 'processing' && (
+              <Message className={style.title} component="h4" content={operationMessage} />
+            )}
+          </>
         ) : (
-          <ErrorNotification small content={error} title={m.errorTitle} />
+          <ErrorNotification
+            small
+            content={error}
+            title={!isFrontend(error) ? m.errorTitle : undefined}
+            messageValues={
+              !isFrontend(error)
+                ? undefined
+                : {
+                    DocLink: docLinkValue,
+                  }
+            }
+          />
         )}
+        {status === 'finished' && (
+          <>
+            {aborted && convertedDevices.length - devicesProcessed !== 0 && (
+              <Notification
+                small
+                warning
+                content={m.abortWarningMessage}
+                title={m.abortWarningTitle}
+                messageValues={{ count: convertedDevices.length - devicesProcessed }}
+              />
+            )}
+            {deviceErrors.length !== 0 ? (
+              <div>
+                {deviceErrors.length >= devicesProcessed ? (
+                  <Notification
+                    small
+                    warning
+                    content={m.noneWarningMessage}
+                    title={m.noneWarningTitle}
+                  />
+                ) : (
+                  <Notification
+                    small
+                    warning
+                    content={m.incompleteWarningMessage}
+                    title={m.incompleteWarningTitle}
+                    messageValues={{ count: deviceErrors.length }}
+                  />
+                )}
+                <Message
+                  component="span"
+                  content={m.incompleteStatus}
+                  values={{ count: deviceErrors.length }}
+                />
+                <ul>
+                  {deviceErrors.map(({ deviceId, error }) => (
+                    <li key={deviceId} className={style.deviceErrorEntry}>
```

Comment:
This might cause the following error:
```
Warning: Encountered two children with the same key, `unknown device ID`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```